gap/weights.gd

```diff
@@ -0,0 +1,17 @@
+#############################################################################
+##
+##  weights.gd
+##  Copyright (C) 2023                                Raiyan Chowdhury
+##
+##  Licensing information can be found in the README file of this package.
+##
+#############################################################################
+##
+
+# 1. Edge Weights
+DeclareAttribute("EdgeWeights", IsDigraph);
+DeclareGlobalFunction("EdgeWeightedDigraph");
+DeclareProperty("IsNegativeEdgeWeightedDigraph", IsDigraph and HasEdgeWeights);
```

Comment:
We might also want to add `IsPositiveEdgeWeightedDigraph` at some point, but not just now!